### PR TITLE
Checking if keyword is a MarkDecorator before set

### PR DESCRIPTION
**Pull request by cr3**
In NodeKeywords, setitem might be called multiple times for the same
key. For example, pytest.mark.foo('foo') will call setitem first with a
MarkDecorator value and then with a True value. The MarkDecorator value is
therefore overwritten and the marked keyword is never matched. This change
only sets the keyword when it's not already a MarkDecorator instance.

### CODE REVIEW

File: _pytest/main.py [hunk 1]
```diff
@@ -210,7 +210,9 @@ def __getitem__(self, key):
             return self.parent.keywords[key]
 
     def __setitem__(self, key, value):
-        self._markers[key] = value
+        from _pytest.mark import MarkDecorator
+        if not isinstance(self._markers.get(key), MarkDecorator):
```
Comment (nicoddemus) [inline _pytest/main.py]: I tested the patch locally and it works as advertised, I'm just not so sure if the fix is correct. @RonnyPfannschmidt has more experience with marks in general, what do you think?

Comment (cr3) [inline _pytest/main.py]: I'm not particularly comfortable with the fix either, so I tried to check for the duplicate key higher in the stack. I looked at `PyCollector._genfunctions`, for example, where I thought that the `keywords` argument passed to `Function` could be empty when `callspec.params['name'] == callspec.params['arg']` but that seemed even more convoluted. Any suggestion would be most welcome.

Comment (RonnyPfannschmidt) [inline _pytest/main.py]: the fix just makes it bad in a different way

 the real problem is, that we track markers in a very horrific manner
its simply the completely wrong datastructure

Comment (nicoddemus) [inline _pytest/main.py]: > the real problem is, that we track markers in a very horrific manner
its simply the completely wrong datastructure

There's no solution then?

Comment (RonnyPfannschmidt) [inline _pytest/main.py]: no, its currently not fixable due to broken internal (i forsee any minimal attempt causing more regressions)

Comment (cr3) [inline _pytest/main.py]: No worries, these thing happen, I'll just close my pull request.

**Pull request closed without merging**